framework/src/problems/FEProblemBase.C

```diff
@@ -3332,64 +3335,86 @@ FEProblemBase::projectSolution()
   FloatingPointExceptionGuard fpe_guard(_app);
 
   ConstElemRange & elem_range = *_mesh.getActiveLocalElementRange();
-  ComputeInitialConditionThread cic(*this);
-  Threads::parallel_reduce(elem_range, cic);
 
-  if (haveFV())
+  // loop over _global_current_state = 2, 1, 0
+  for (_global_current_state = 1; _global_current_state >= 0; _global_current_state--)
```

Comment:
But this is not 2, 1, 0 -- this is 1, 0. Now I wonder if you have test coverage for checking the older IC being set correctly...

Reply:
I don't. I shall add some.